app/src/main/java/ru/woodymsk/socialapp/presentation/map_screen/mapper/ObjectDetailsMapper.kt

```diff
@@ -0,0 +1,54 @@
+package ru.woodymsk.socialapp.presentation.map_screen.mapper
+
+import com.yandex.mapkit.GeoObject
+import com.yandex.mapkit.search.BusinessObjectMetadata
+import com.yandex.mapkit.search.ToponymObjectMetadata
+import com.yandex.mapkit.uri.UriObjectMetadata
+import ru.woodymsk.socialapp.R
+import ru.woodymsk.socialapp.presentation.common.ResourcesService
+import ru.woodymsk.socialapp.presentation.map_screen.model.ObjectDetails
+import ru.woodymsk.socialapp.presentation.map_screen.model.TypeSpecificDetails
+import javax.inject.Inject
+
+
+class ObjectDetailsMapper @Inject constructor(
+    private val resourcesService: ResourcesService,
+) {
+
+    fun mapGeoObjectToObjectDetails(geoObject: GeoObject?): ObjectDetails {
+        val uri =
+            geoObject?.metadataContainer?.getItem(UriObjectMetadata::class.java)?.uris?.firstOrNull()?.value
+
+        val typeSpecificDetails =
+            geoObject?.metadataContainer?.getItem(ToponymObjectMetadata::class.java)?.let { toponym ->
+                TypeSpecificDetails.Toponym(
+                    address = toponym.address.formattedAddress,
+                    components = toponym.address.components.joinToString { it.name }
+                )
+            } ?: geoObject?.metadataContainer?.getItem(BusinessObjectMetadata::class.java)
+                ?.let { business ->
+                    TypeSpecificDetails.Business(
+                        name = business.name,
+                        workingHours = business.workingHours?.text,
+                        categories = business.categories.map { it.name }.takeIf { it.isNotEmpty() }?.toSet()
+                            ?.joinToString(),
```

Comment:
Я такие вещи вот так предпочитаю писать, чтобы было видно все преобразования:
```suggestion
                        categories = business.categories.map { it.name }
                            .takeIf { it.isNotEmpty() }
                            ?.toSet()
                            ?.joinToString(),
```

Reply:
да, так читабельнее, поправил во всём классе

Reply:
неа

Reply:
это просто не запушил. Там теперь по-другому